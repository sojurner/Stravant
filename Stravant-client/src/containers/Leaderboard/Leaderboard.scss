table {
  border-collapse: collapse;
  width: 100%;
}

th,
td {
  text-align: left;
  padding: 8px;
}

tr:nth-child(even) {
  background-color: #f2f2f2;
}

tr:nth-child(odd):hover {
  background-color: #c6b7ba;
}

tr:nth-child(even):hover {
  background-color: #dadada;
  box-shadow: 3px 3px 5px #aaaaaa;
}
