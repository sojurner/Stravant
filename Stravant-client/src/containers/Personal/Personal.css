.personal-stats {
  display: flex;
  justify-content: center; }

path {
  cursor: pointer; }

.personal-stats {
  position: absolute; }

.legend {
  background-color: #f9f9f9;
  border: 1px solid #e5e5e5;
  border-radius: 12px;
  font-size: 0.8em;
  padding: 2rem; }

.pie-chart-wrap {
  display: flex;
  justify-content: center; }

li {
  display: flex;
  left: 0.6rem; }

.data-details {
  position: absolute;
  top: 12rem;
  left: 23rem;
  font-family: daybold;
  font-size: 1.5rem;
  color: #ffc1b8;
  letter-spacing: 2px;
  font-size: 1.5rem;
  background: #c5696a;
  border-radius: 0.4em;
  padding: 1rem;
  opacity: 0.9; }

.data-details:after {
  content: '';
  position: absolute;
  left: 0;
  top: 50%;
  width: 0;
  height: 0;
  border: 21px solid transparent;
  border-right-color: #c5696a;
  border-left: 0;
  border-bottom: 0;
  margin-top: -10.5px;
  margin-left: -21px;
  opacity: 0.9; }
