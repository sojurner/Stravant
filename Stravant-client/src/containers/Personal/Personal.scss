.personal-stats {
  display: flex;
  justify-content: center;
}

path {
  cursor: pointer;
}

.personal-stats {
  position: absolute;
}

.legend {
  background-color: #f9f9f9;
  border: 1px solid #e5e5e5;
  border-radius: 12px;
  font-size: 0.8em;
  padding-top: 2rem;
}

.running-avg {
  text-align: center;
  font-family: daybold;
  font-size: 2.5rem;
  padding: auto;
  color: #a49188;
  margin: 0;
}

.pie-chart-wrap {
  display: flex;
  justify-content: center;
}

li {
  display: flex;
  left: 0.6rem;
}

.data-details {
  position: fixed;
  font-family: daybold;
  font-size: 1.5rem;
  color: #ffc1b8;
  letter-spacing: 2px;
  background: #c5696a;
  border-radius: 0.4em;
  margin: 26rem 0;
  padding: 1rem;
  opacity: 0.9;
}

.legend-container-fc25d28b8dda2e2172d549eafe0de054a26a4837 {
  width: 11rem;
  margin: -3rem 0 0 -23rem;
  position: fixed;
  top: 37%;
  left: 50%;
  padding: 0.5rem 0 0 0;
  font-family: daybold;
  font-size: 1.5rem;
}

.legend {
  padding-top: 0.7rem;
  background-color: #b3aaab;
}

.li {
  padding-top: 0.4rem;
}

.spinning-globe {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  margin: 3rem auto;
  background: center/cover url(../../images/earth.png);
  box-shadow: inset 0 0 25px #b07979;
  animation: world 35s linear infinite;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #a49188;
}

.spinning-globe:hover {
  box-shadow: inset 0 0 25px #8b4242;
}

@keyframes world {
  0% {
    background-position: 0 0;
  }
  100% {
    background-position: 719px 0;
  }
}

@media screen and (max-width: 800px) {
  .legend-container-fc25d28b8dda2e2172d549eafe0de054a26a4837 {
    width: 8.3rem;
    margin: -3rem 0 0 5rem;
    top: 85%;
    left: 0%;
    padding: 0.5rem 0 0 0;
    font-family: daybold;
    font-size: 1rem;
    z-index: 1;
  }
}

@media screen and (max-width: 510px) {
  .pie-chart-wrap {
    margin-top: 1.4rem;
  }

  .running-avg {
    margin: 0.7rem;
  }

  .data-details {
    margin: 27.5rem 0 0 0;
  }

  .pie-chart-7450ce7f65e862c33c80040ea675f203302df7d5 {
    position: absolute;
    left: 2rem;
  }
}
