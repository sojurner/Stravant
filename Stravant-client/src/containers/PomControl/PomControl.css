.pom-instruction,
.stop-instruction {
  position: absolute;
  bottom: 38%;
  left: 47.5%;
  font-size: 1.2rem;
  font-family: daybold;
  color: azure;
  margin: -0.6rem 0 0 -0.6rem; }

.track-poms {
  font-family: daybold;
  font-size: 3rem;
  margin: 0.5rem 0 0 0;
  text-align: center;
  color: #d19e86; }

img {
  cursor: pointer; }

i {
  cursor: pointer; }

.pom-details {
  text-align: center; }

.socket-msg {
  position: absolute;
  font-size: 2.5rem;
  font-family: daybold;
  margin: -12.5rem 0 0 -12.5rem;
  left: 50%;
  width: 25rem;
  top: 30%;
  border-radius: 5px;
  background: #b19494; }

.modal-timer {
  z-index: 1;
  background-color: #c2b8b5;
  padding: 1rem;
  text-align: center; }

.fa-play-circle {
  color: #c2bcba;
  font-size: 4rem;
  margin: 5rem 2rem 1.5rem 2rem; }

.fa-stop-circle {
  color: #d19e86;
  font-size: 4rem;
  margin: 1rem 2rem; }

.pom-control {
  margin: auto;
  display: inline-block; }

.fa-stop-circle:hover {
  color: #833434;
  text-align: right; }

.modal {
  display: inline-block; }

.fa-play-circle:hover {
  color: #979290; }

.time {
  z-index: 1;
  background: #b19494;
  border: 2px dotted black;
  position: absolute;
  padding: 3rem 1.5rem;
  left: 48%;
  margin: -12.5rem 0 0 -12.5rem;
  top: 42%;
  font-family: daybold;
  font-size: 4.5rem;
  animation: fadein 1s;
  border-radius: 8px;
  max-width: 25rem;
  min-width: 25rem; }

.summary {
  position: absolute;
  animation: fadein 0.5s;
  font-family: daybold;
  font-size: 2rem;
  left: 49%;
  top: 55%;
  margin: -5rem 0 0 -5rem;
  color: #c2bcba; }

.abbr {
  font-size: 2rem;
  display: inline-block; }

@keyframes fadein {
  from {
    opacity: 0; }
  to {
    opacity: 1; } }

i {
  font-size: 2rem; }

@media screen and (max-width: 510px) {
  .time {
    top: 37.3%;
    left: 44%; }
  .pom-instruction,
  .stop-instruction {
    bottom: 51%;
    left: 44%;
    font-size: 1.2rem;
    font-family: daybold;
    color: azure; }
  .track-poms {
    margin: 2rem 0 0 0;
    font-size: 2rem; }
  .summary {
    left: 47%;
    top: 41%;
    margin: -5rem 0 0 -5rem; } }

@media screen and (max-height: 668px) {
  .summary {
    left: 47%;
    top: 50%;
    margin: -5rem 0 0 -5rem; } }
