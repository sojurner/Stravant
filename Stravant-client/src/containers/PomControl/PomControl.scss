.pom-instruction {
}

img {
  cursor: pointer;
}
i {
  cursor: pointer;
}

.modal-timer {
  z-index: 1;
  background-color: #fed2bf;
  padding: 1rem;
  position: absolute;
  text-align: center;
  animation: fadein 1s;
}

.fa-hand-paper {
  font-size: 4rem;
}

.fa-hand-paper:hover {
  color: rgb(131, 52, 52);
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

i {
  font-size: 2rem;
}
