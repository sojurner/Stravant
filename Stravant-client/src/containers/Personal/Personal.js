import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PieChart, Legend, ToolTip } from 'react-easy-chart';
import * as apiFetch from '../../helpers/apiCalls/apiCalls';
import * as userActions from '../../actions/userAction';
import './Personal.css';
import { type } from 'os';

export class Personal extends Component {
  constructor() {
    super();
    this.state = {
      showToolTip: false,
      top: '',
      left: '',
      value: '',
      key: '',
      goal: 0
    };
  }

  componentDidMount() {
    this.getUserStats();
  }

  getUserStats = async () => {
    const { setTotalStats } = this.props;
    const { accessToken, userId } = this.props.currentUser.info;
    const userStats = await apiFetch.getAggregateStats(accessToken, userId);
    setTotalStats(userStats);
  };

  mouseOverHandler = data => {
    console.log(data);
    this.setState({
      key: data.data.key,
      value: data.data.value,
      showToolTip: true
    });
  };

  mouseOutHandler = () => {
    this.setState({ showToolTip: false });
  };

  render() {
    const { totalStats } = this.props.currentUser;
    const filteredActivity = Object.keys(totalStats).map(type => {
      const destructured = totalStats[type];
      let dataArr = [];
      let configArr = [];
      Object.keys(destructured).forEach((stat, index) => {
        const filtered = {
          key: stat,
          value: Math.round(destructured[stat] * 100) / 100,
          color: `rgb(${index}04, 1${index}9, 2${index}4)`
        };

        const config = { color: `rgb(${index}04, 1${index}9, 2${index}4)` };
        dataArr.push(filtered);
        configArr.push(config);
      });
      return (
        <div>
          <PieChart
            className="personal-piechart"
            data={dataArr}
            innerHoleSize={150}
            mouseOverHandler={this.mouseOverHandler}
            mouseOutHandler={event => this.mouseOutHandler(event)}
            padding={10}
          />
          <Legend data={dataArr} config={configArr} dataId={'key'} />
        </div>
      );
    });

    return (
      <div className="pie-chart-wrap">
        {filteredActivity}
        {this.state.value &&
          this.state.showToolTip && (
            <p className="data-details">
              {`${this.state.key}: ${this.state.value}`}
            </p>
          )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.currentUser
});

const mapDispatchToProps = dispatch => ({
  setTotalStats: stats => dispatch(userActions.setTotalStats(stats))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Personal);
