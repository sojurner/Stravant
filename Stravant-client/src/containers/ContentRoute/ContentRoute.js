import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import Weekly from '../Weekly/Weekly';
import Personal from '../Personal/Personal';
import Leaderboard from '../Leaderboard/Leaderboard';
import PomControl from '../../containers/PomControl/PomControl';

export class ContentRoute extends Component {
  render() {
    return (
      <Switch>
        <Route exact path="/" />
        <Route exact path="/personal" component={Personal} />
        <Route exact path="/weekly" component={Weekly} />
        <Route exact path="/leaderboard" component={Leaderboard} />
        <Route exact path="/pomodoro" component={PomControl} />
      </Switch>
    );
  }
}

export default ContentRoute;
