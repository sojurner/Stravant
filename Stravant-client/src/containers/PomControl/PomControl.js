import React, { Component } from 'react';
import { connect } from 'react-redux';
import openSocket from 'socket.io-client';
import moment from 'moment';
// import { createActivity } from '../../helpers/apiCalls/apiCalls';
import { PomContainer } from '../../components/PomContainer/PomContainer';
import * as pomActions from '../../actions/pomAction';
import { getLastPomTime } from '../../helpers/helpers/helpers';

import './PomControl.css';

var socket = openSocket('http://localhost:5000');
export class PomControl extends Component {
  constructor() {
    super();
    this.state = {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      start: false,
      stop: true,
      save: false,
      hide: false,
      description: false,
      pomSummary: '',
      socketMessage: ''
    };
  }

  componentDidMount() {
    const { setPomHistory } = this.props;
    if (localStorage.getItem('pomHistory')) {
      const storageItem = JSON.parse(localStorage.getItem('pomHistory'));
      setPomHistory(storageItem);
    }
  }

  togglePom = (event, string) => {
    const { start, hide } = this.state;
    if (start) {
      const toggleHide = !hide;
      this.setState({ hide: toggleHide });
    } else {
      this.sendSocketMsg(event);
      this.setState({ [string]: true, stop: false });
    }
    const { togglePomState, pomInfo } = this.props;
    togglePomState(!pomInfo.pomStatus);
  };

  sendSocketMsg = () => {
    const { firstName } = this.props.currentUser.info;
    socket.emit('pomStart', `${firstName} just started on Pom!`);
  };

  startMSecond = () => {
    let time = this.state.mSecond + 1;
    time < 10
      ? this.setState({ mSecond: time })
      : this.setState({ mSecond: 0 });
  };

  startSecond = () => {
    let time = this.state.second + 1;
    time < 60 ? this.setState({ second: time }) : this.setState({ second: 0 });
  };

  startMinute = () => {
    let time = this.state.minute + 1;
    time < 60 ? this.setState({ minute: time }) : this.setState({ minute: 0 });
  };

  startHour = () => {
    let time = this.state.hour + 1;
    this.setState({ hour: time });
  };

  stopTime = () => {
    const { firstName } = this.props.currentUser.info;
    const { pomHistory } = this.props.pomInfo;
    const timeKeys = Object.keys(pomHistory);
    const lastPom = timeKeys[timeKeys.length - 1];
    const currentTime = moment().format('llll');
    if (lastPom) {
      const difference = getLastPomTime(currentTime, lastPom);
      socket.emit(
        'pomEnd',
        `${firstName} finished the Pom!  Your last pom was ${difference}`
      );
    }
    this.setState({ start: false, stop: true });
  };

  addDescription = string => {
    this.setState({ description: string });
  };

  removeDescription = () => {
    this.setState({ description: '' });
  };

  resetTimer = () => {
    const { setPomHistory } = this.props;
    const { minute, second, hour } = this.state;
    const timeState = {
      second: second,
      minute: minute,
      hour: hour
    };
    if (localStorage.getItem('pomHistory')) {
      const storageItem = JSON.parse(localStorage.getItem('pomHistory'));
      const newItem = {
        ...storageItem,
        [moment().format('llll')]: timeState
      };
      localStorage.setItem('pomHistory', JSON.stringify(newItem));
    } else {
      const itemToStorage = {
        [moment().format('llll')]: timeState
      };
      localStorage.setItem('pomHistory', JSON.stringify(itemToStorage));
    }
    const storageItem = JSON.parse(localStorage.getItem('pomHistory'));
    setPomHistory(storageItem);

    const timeSummary = `${second}s  ${minute}m  ${hour}h`;
    this.setState({
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      start: false,
      save: true,
      pomSummary: timeSummary
    });
  };

  removePom = (history, time) => {
    delete history[time];
    this.props.setPomHistory(history);
    localStorage.setItem('pomHistory', JSON.stringify(history));
  };

  render() {
    socket.on('pomClick', msg => {
      this.setState({ socketMessage: msg });
    });
    const { pomHistory } = this.props.pomInfo;
    const styles = {
      pomStyle: {
        position: 'absolute',
        left: window.screenX + 10,
        top: window.screenY - 20
      },

      summaryStyle: {
        position: 'absolute',
        left: window.screenX + 25,
        top: window.screenY + 90
      },

      suggestStyle: {
        position: 'absolute',
        left: window.screenX + 25,
        top: window.screenY + 60
      },

      modalStyle: {
        position: 'absolute',
        left: window.screenX + 100,
        top: window.screenY - 20
      }
    };
    const {
      mSecond,
      second,
      minute,
      hour,
      start,
      stop,
      description,
      hide,
      save,
      pomSummary,
      socketMessage
    } = this.state;
    return (
      <div>
        {(second > 0 || mSecond > 0 || hour > 0 || minute > 0) &&
          !hide &&
          start && (
            <section className="modal" style={styles.modalStyle}>
              <div className="time">
                <span className="time hour">{hour} h</span>
                <span className="time minute">
                  ..
                  {minute} m
                </span>
                <span className="time second">
                  ..
                  {second} s
                </span>
                <span className="time mSecond">
                  ..
                  {mSecond} ms
                </span>
              </div>
              <div className="save-stop">
                <i
                  class="far fa-hand-paper"
                  onClick={() => {
                    this.stopTime();
                    if (!stop) {
                      clearInterval(this.mSeconds);
                      clearInterval(this.seconds);
                      clearInterval(this.minutes);
                      clearInterval(this.hours);
                      this.resetTimer();
                    }
                  }}
                  onMouseEnter={() => this.addDescription('stop')}
                  onMouseLeave={this.removeDescription}
                />
              </div>
            </section>
          )}
        {save && (
          <h4 style={styles.summaryStyle}>
            <div>Most recent:</div>
            ---
            {pomSummary}
            ---
          </h4>
        )}
        {socketMessage && <h4>{socketMessage}</h4>}
        <img
          src={require('../../images/pomodoro-icon.png')}
          style={styles.pomStyle}
          height="100px"
          width="100px"
          className="pomodoro"
          onMouseEnter={() => this.addDescription('start')}
          onMouseLeave={this.removeDescription}
          onClick={event => {
            this.togglePom(event, 'start');
            if (!start && stop) {
              this.mSeconds = setInterval(this.startMSecond, 101);
              this.seconds = setInterval(this.startSecond, 1001);
              this.minutes = setInterval(this.startMinute, 60001);
              this.hours = setInterval(this.startHour, 3600001);
            }
          }}
        />
        {!pomHistory && <p>You have no record Poms</p>}
        {description === 'start' &&
          !hide &&
          !start && (
            <p style={styles.suggestStyle} className="pom-instruction">
              Start Pom?
            </p>
          )}
        {description === 'start' &&
          !hide &&
          start && (
            <p style={styles.suggestStyle} className="pom-instruction">
              Hide Pom?
            </p>
          )}
        {description === 'stop' && (
          <p style={styles.suggestStyle} className="stop-instruction">
            End Pom?
          </p>
        )}

        {hide && (
          <p style={styles.suggestStyle} className="stop-instruction">
            Show Pom?
          </p>
        )}
        <PomContainer removePom={this.removePom} pomHistory={pomHistory} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.currentUser,
  pomInfo: state.pomInfo
});

const mapDispatchToProps = dispatch => ({
  setPomHistory: history => dispatch(pomActions.setPomHistory(history)),
  togglePomState: bool => dispatch(pomActions.togglePomState(bool))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PomControl);
