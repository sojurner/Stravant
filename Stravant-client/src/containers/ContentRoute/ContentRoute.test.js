import React from 'react';
import { shallow } from 'enzyme';
import ContentRoute from './ContentRoute';
import Weekly from '../Weekly/Weekly';
import { Personal } from '../Personal/Personal';
import Leaderboard from '../Leaderboard/Leaderboard';
import { PomControl } from '../PomControl/PomControl';

describe('ContentRoute', () => {
  it('should matchsnapshot when path matches personal', () => {
    let personalWrapper = shallow(
      <ContentRoute path="/personal" component={Personal} />
    );
    expect(personalWrapper).toMatchSnapshot();
  });

  it('should MatchSnapshot when path matches leaderboard', () => {
    let leaderboardWrapper = shallow(
      <ContentRoute path="/leaderboard" component={Leaderboard} />
    );
    expect(leaderboardWrapper).toMatchSnapshot();
  });

  it('should MatchSnapshot when path matches Compare', () => {
    let compareWrapper = shallow(
      <ContentRoute path="/weekly" component={Weekly} />
    );
    expect(compareWrapper).toMatchSnapshot();
  });

  it('should MatchSnapshot when path matches Pomodoro', () => {
    let pomWrapper = shallow(
      <ContentRoute path="/pomodoro" component={PomControl} />
    );

    expect(pomWrapper).toMatchSnapshot();
  });
});
