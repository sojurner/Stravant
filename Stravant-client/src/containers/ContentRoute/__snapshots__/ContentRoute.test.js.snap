// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentRoute should MatchSnapshot when path matches Compare 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/personal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/weekly"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/leaderboard"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/pomodoro"
  />
</Switch>
`;

exports[`ContentRoute should MatchSnapshot when path matches Pomodoro 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/personal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/weekly"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/leaderboard"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/pomodoro"
  />
</Switch>
`;

exports[`ContentRoute should MatchSnapshot when path matches leaderboard 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/personal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/weekly"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/leaderboard"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/pomodoro"
  />
</Switch>
`;

exports[`ContentRoute should matchsnapshot when path matches personal 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/personal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/weekly"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/leaderboard"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/pomodoro"
  />
</Switch>
`;
